src/main/java/org/zalando/pazuzu/api/FeatureServiceRestImpl.java

```diff
@@ -0,0 +1,205 @@
+package org.zalando.pazuzu.api;
+
+import io.swagger.annotations.ApiParam;
+import io.swagger.api.ApiApi;
+import io.swagger.api.NotFoundException;
+import io.swagger.model.Feature;
+import io.swagger.model.FeatureMeta;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import org.zalando.pazuzu.exception.FeatureNotFoundException;
+import org.zalando.pazuzu.feature.FeatureService;
+import org.zalando.pazuzu.feature.FeaturesWithTotalCount;
+import org.zalando.pazuzu.security.Roles;
+
+import javax.annotation.security.RolesAllowed;
+import javax.transaction.Transactional;
+import java.net.URI;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Created by hhueter on 16/01/2017.
+ */
+@RestController
+public class FeatureServiceRestImpl implements ApiApi {
+    private static final String X_TOTAL_COUNT = "X-Total-Count";
+    private final FeatureService featureService;
+
+    private static ThreadLocal<DateFormat> dateFormat = new ThreadLocal<DateFormat>() {
+        @Override
+        protected DateFormat initialValue() {
+            return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        }
+    };
+
+    @Autowired
+    public FeatureServiceRestImpl(FeatureService featureService) {
+        this.featureService = featureService;
+    }
+
+    @Override
+    @RolesAllowed({Roles.ANONYMOUS, Roles.USER})
+    public Callable<ResponseEntity<List<Feature>>> apiFeaturesGet(@ApiParam(value = "value, that must present in feature name.") @RequestParam(value = "names", required = false) List<String> names, @ApiParam(value = "the offset to start from.") @RequestParam(value = "offset", required = false) Integer offset, @ApiParam(value = "maximum number of features to return.") @RequestParam(value = "limit", required = false) Integer limit) throws NotFoundException {
```

Comment:
The method declaration is too long to fit in one line and is difficult to read in Github.
Having the different parameters in different lines will probably help.